src/component/gestures.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import Hammer from 'react-hammerjs';
+
+const { object } = React.PropTypes;
+const hammerOptions = {
+  recognizers: {
+    pan: {
+      threshold: 0,
+    },
+  },
+};
+
+export default class Gestures extends React.Component {
+  static propTypes = {
+    options: object,
+  }
+
+  componentWillMount() {
+    this.wrapProps(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.wrapProps(nextProps);
+  }
+
+  wrapProps(props) {
+    const wrapped = {};
+    const propNames = Object.keys(props);
+    const deltaEvents = propNames.filter(name => /^on(Pan)$/.test(name));
+    const endEvents = propNames.map(name => `${name}End`);
+
+    deltaEvents.forEach(name => {
+      wrapped[name] = e => this.handleDeltaEvent(e, name, props[name]);
+    });
```

Comment:
Problematic - will create new functions when receiving props. Should check if already there. Should delete handlers when props are gone.